File: CS/Helpers/DiagramService.cs
using DevExpress.Mvvm.UI;
using DevExpress.Xpf.Diagram;
using System.IO;
using System.Text;

namespace MDI_Diagram {
    public interface IDiagramService {
        void Restore(string diagram);
        string Save();
    }

    public class DiagramService : ServiceBase, IDiagramService {
        DiagramControl diagram => AssociatedObject as DiagramControl;

        public void Restore(string layout) {
            var stream = new MemoryStream(Encoding.UTF8.GetBytes(layout));
            diagram?.LoadDocument(stream);
        }

        public string Save() {
            var stream = new MemoryStream();
            diagram?.SaveDocument(stream);
            stream.Seek(0, SeekOrigin.Begin);
            return Encoding.UTF8.GetString(stream.ToArray());
        }
    }
}

File: VB/Helpers/DiagramService.vb
Imports DevExpress.Mvvm.UI
Imports DevExpress.Xpf.Diagram
Imports System.IO
Imports System.Text

Namespace MDI_Diagram

    Public Interface IDiagramService

        Sub Restore(ByVal diagram As String)

        Function Save() As String

    End Interface

    Public Class DiagramService
        Inherits ServiceBase
        Implements IDiagramService

        Private ReadOnly Property diagram As DiagramControl
            Get
                Return TryCast(AssociatedObject, DiagramControl)
            End Get
        End Property

        Public Sub Restore(ByVal layout As String) Implements IDiagramService.Restore
            Dim stream = New MemoryStream(Encoding.UTF8.GetBytes(layout))
            diagram?.LoadDocument(stream)
        End Sub

        Public Function Save() As String Implements IDiagramService.Save
            Dim stream = New MemoryStream()
            diagram?.SaveDocument(stream)
            stream.Seek(0, SeekOrigin.Begin)
            Return Encoding.UTF8.GetString(stream.ToArray())
        End Function
    End Class
End Namespace
